### PR TITLE
Do not enforce basic keyboard X11Variant

### PR DESCRIPTION
**Pull request by ancorgs**
See https://bugzilla.suse.com/show_bug.cgi?id=1020082#c8 for the rationale.

### CODE REVIEW

File: keyboard/src/modules/Keyboard.rb [hunk 1]
```diff
@@ -1551,6 +1539,23 @@ def all_keyboards
       # eval is necessary for translating the texts needed to be translated
       content ? Builtins.eval(content) : {}
     end
+
+    def call_set_x11_keymap
+      args = [@XkbLayout, @XkbModel, @XkbVariant, @XkbOptions]
+      return if args.all?(&:empty?)
+      
+      # According to localectl syntax, empty options must be entered as "" if
+      # there are more non-empty options afterwards and skipped otherwise
```
Comment (mchf) [inline keyboard/src/modules/Keyboard.rb]: I had to read this several times ... and still didn't get it until I read the code ;-)

File: keyboard/src/modules/Keyboard.rb [hunk 2]
```diff
@@ -1551,6 +1539,24 @@ def all_keyboards
       # eval is necessary for translating the texts needed to be translated
       content ? Builtins.eval(content) : {}
     end
+
+    def call_set_x11_keymap
+      args = [@XkbLayout, @XkbModel, @XkbVariant, @XkbOptions]
+      return if args.all?(&:empty?)
```
Comment (mchf) [inline keyboard/src/modules/Keyboard.rb]: i guess that none of ```@Xkb....``` can be nil.

Comment (ancorgs) [inline keyboard/src/modules/Keyboard.rb]: No, it cannot. They are all initialized to a string (to `""` in most cases).

Comment (mchf) [inline keyboard/src/modules/Keyboard.rb]: i'll answer myself ... no it cannot

**Pull request merged**